### PR TITLE
Delete push repositories with their distribution

### PR DESCRIPTION
**Pull request by mdellweg**
Both entities can be used to trigger the deletion, and the deletions are
coordinated in an atomic transaction.

fixes #7172
https://pulp.plan.io/issues/7172

### CODE REVIEW

File: pulp_container/app/viewsets.py [hunk 1]
```diff
@@ -433,6 +455,30 @@ class ContainerDistributionViewSet(BaseDistributionViewSet):
...
+    def destroy(self, request, pk, **kwargs):
+        """
+        Delete a push repository with its distribution
+        """
+        distribution = self.get_object()
+        reservations = [distribution]
+        instance_ids = [
+            (distribution.pk, "container", "ContainerDistributionSerializer"),
+        ]
+        if distribution.repository and distribution.repository.PUSH_ENABLED:
```
Comment (ipanova) [inline pulp_container/app/viewsets.py:471]: ```AttributeError: 'Repository' object has no attribute 'PUSH_ENABLED'```

You need to use `cast()`

File: pulp_container/app/tasks/base.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+from django.db import transaction
+
+from pulpcore.app.apps import get_plugin_config
+
+
+def general_multi_delete(instance_ids):
+    """
+    Delete a list of model in a transaction
```
Comment (ipanova) [inline pulp_container/app/tasks/base.py:8]: s/model/models/

Comment (mdellweg) [reply pulp_container/app/tasks/base.py:8]: Wdyt about "model instances"?

File: pulp_container/app/viewsets.py [hunk 3]
```diff
@@ -433,6 +455,30 @@ class ContainerDistributionViewSet(BaseDistributionViewSet):
     queryset = models.ContainerDistribution.objects.all()
     serializer_class = serializers.ContainerDistributionSerializer
 
+    @extend_schema(
+        description="Trigger an asynchronous delete task",
+        responses={202: AsyncOperationResponseSerializer},
+    )
+    def destroy(self, request, pk, **kwargs):
+        """
+        Delete a push repository with its distribution
```
Comment (ipanova) [inline pulp_container/app/viewsets.py:464]: Delete a distribution. If a push repository is associated to it, delete it as well.

**Pull request merged**